src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs

```diff
@@ -203,6 +204,32 @@ public override string ToString()
 ";
 
         [Fact]
+        [WorkItem(14844, "https://github.com/dotnet/roslyn/issues/14844")]
+        public void InterfaceImplAttributesAreNotSharedAcrossTypeRefs()
```

Comment:
@AlekseyTs I don't see a way to make a similar test for constraints, since there's no error for a substituted type parameter and the constraint having differing tuple names.

Could you give me a hint on how to construct such a test? :)